src/docfx/restore/Restore.cs

```diff
@@ -77,16 +126,38 @@ public static (string restoreDir, string url, string refSpec) GetGitRestoreInfo(
                     if (GitUtility.IsRepo(restoreDir))
                     {
                         // already exists, just pull the new updates from remote
-                        await GitUtility.Pull(restoreDir);
+                        await GitUtility.Fetch(restorePath);
                     }
                     else
                     {
                         // doesn't exist yet, clone this repo to a specified branch
-                        await GitUtility.Clone(restoreDir, url, restoreDir, rev);
+                        await GitUtility.Clone(restoreDir, url, restorePath, null, true);
+                    }
+
+                    workTreeHead = await GitUtility.Revision(restorePath, rev);
+                    workTreePath = GetWorkTreePath(restoreDir, workTreeHead);
+                    var workTrees = await GitUtility.ListWorkTrees(restorePath);
```

Comment:
>ListWorkTrees [](start = 53, length = 13)

`git worktree add` exits with code 128 if the worktree already exists. can we use that over `ListWorkTree` ?

Reply:
I don't want to depends on git command error code right now

---
In reply to: [200891920](https://github.com/dotnet/docfx/pull/3020#discussion_r200891920) [](ancestors = 200891920)

Reply:
128 is really a general error code

---
In reply to: [200893630](https://github.com/dotnet/docfx/pull/3020#discussion_r200893630) [](ancestors = 200893630,200891920)